### ISSUE TITLE
Selection doesn't work correctly, if we start to select from the beginning of the line

### ISSUE THREAD

**Issue by igordmn** [JetBrains/compose-jb#1425]
Windows 11, compose 1.0.0-beta6-build464

https://user-images.githubusercontent.com/5963351/142635296-b5993427-125b-4ef9-bd30-9c824c46d07e.mp4

```
import androidx.compose.foundation.text.selection.SelectionContainer
import androidx.compose.material.Text
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.unit.sp
import androidx.compose.ui.window.singleWindowApplication

fun main() = singleWindowApplication {
    SelectionContainer {
        Text(text = (1..100).joinToString { "Text " }, style = TextStyle(fontSize = 50.sp))
    }
}
```

**Comment by igordmn** [comment]
In this code:
```
    override fun getHorizontalPosition(offset: Int, usePrimaryDirection: Boolean): Float {
        val prevBox = getBoxBackwardByOffset(offset)
        val nextBox = getBoxForwardByOffset(offset)
```

we choose `prevBox`, which corresponds to the previous line symbol (the last space in our case)

**Comment by igordmn** [comment]
Because of this, we have a bug in TextField too:

https://user-images.githubusercontent.com/5963351/142664765-01dde90f-49e8-4e1e-93ec-c39426a09a97.mp4

**Comment by akurasov** [comment]
Fixed in 1.0.0-rc2

**Comment by okushnikov** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/CMP-1425">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.
